scipy/io/mmio.py

```diff
@@ -739,22 +748,21 @@ def _write(self, stream, a, comment='', field=None, precision=None,
             # write shape spec
             stream.write(asbytes('%i %i %i\n' % (rows, cols, coo.nnz)))
 
-            # make indices and data array
+            template = self._field_template(field, precision-1)
+
             if field == self.FIELD_PATTERN:
-                IJV = vstack((coo.row, coo.col)).T
-            elif field in [self.FIELD_INTEGER, self.FIELD_REAL]:
-                IJV = vstack((coo.row, coo.col, coo.data)).T
+                for r, c in zip(coo.row+1, coo.col+1):
+                    stream.write(asbytes("%i %i\n" % (r, c)))
+            elif field in (self.FIELD_INTEGER, self.FIELD_REAL):
+                for r, c, d in zip(coo.row+1, coo.col+1, coo.data):
+                    stream.write(asbytes(("%i %i " % (r, c)) +
+                                         (template % d)))
             elif field == self.FIELD_COMPLEX:
-                IJV = vstack((coo.row, coo.col, coo.data.real,
-                              coo.data.imag)).T
+                for r, c, d in zip(coo.row+1, coo.col+1, coo.data):
+                    stream.write(asbytes(("%i %i " % (r, c)) +
+                                         (template % (d.real, d.imag))))
             else:
                 raise TypeError('Unknown field type %s' % field)
-            IJV[:, :2] += 1  # change base 0 -> base 1
-
-            # formats for row indices, col indices and data columns
-            fmt = ('%i', '%i') + ('%%.%dg' % precision,) * (IJV.shape[1]-2)
-            # save to file
-            savetxt(stream, IJV, fmt=fmt)
 
 
 def _is_fromfile_compatible(stream):
```

Comment:
If code isn't used, then yes we prefer that it's deleted rather than sits around. Maybe remove it once the reviewers agree that the approach in this PR is good.